src/com/amazon/corretto/crypto/provider/EdGen.java

```diff
@@ -18,8 +19,10 @@ class EdGen extends KeyPairGeneratorSpi {
   }
 
   public void initialize(int keysize, SecureRandom random) {
-    // Has some behavior in Java, but throws error as placeholder for now.
-    throw new UnsupportedOperationException();
+    if (keysize != 255) {
+      throw new InvalidParameterException("Params must be Ed25519.");
+    }
+    return;
```

Comment:
This needs to call the `initialize(AlgorithmParameterSpec params, SecureRandom random)` interface. The idea is that some folks don't necessarily care/know what to put in AlgorithmParameterSpec. This interface is a simpler one which allows them to use the algorithm with sensible parameters selected by the maintainer. The way you have it right now, a client can call this interface and end up with an uninitialized KeyPairGenerator.

See the javadoc [here](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/KeyPairGeneratorSpi.html#initialize(int,java.security.SecureRandom)

Reply:
Spoke offline to clear this up. This implementation does not need to be initialized and we do not support customizing the RNG. The JDK implementation of edwards curves does accept an AlgorithmParameterSpec which allows the user to select between ed448/ed25519 but we don't support that either. We considered using reflections to provide compatibility for this, but there's a lot more work involved for us to support drop-in capability with JDK.

Reply:
Can you remove the if statement here and simply throw an UnsupportedException like you had originally? After that, I'd be happy to approve the PR.

Reply:
Just updated.